### ISSUE TITLE
Fear + Join to arena (BB #605)

### ISSUE THREAD

**Issue by OregonAsari** [OregonCore/OregonCore#605]
This issue was migrated from bitbucket.
**Original Reporter:** smoldar
**Original Date:** 09.07.2014 20:12:29 GMT+0000
**Original Priority:** major
**Original Type:** bug
**Original State:** closed
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/605

<hr>

When someone join to arena and at pvp zone get fear from other player, sometimes player fall down under map in arena preparation.

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** Zaffy
**Original Date:** 09.07.2014 20:25:53 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/605/#comment-11156609

<hr>

Are you sure ? All debuffs are dropped when you get into an arena

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** smoldar
**Original Date:** 10.07.2014 08:23:56 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/605/#comment-11165934

<hr>

Yes at 3 times i was spamed by my friends, that this problem was created. Fear i think is debuffed, but i think about falling under map when porting to arena with fear. It must change some coords...

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** LordUsagi
**Original Date:** 10.07.2014 15:02:01 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/605/#comment-11173782

<hr>

```
#!c++

# HG changeset patch
# User LordUsagi
# Date 1405004398 10800
#      Thu Jul 10 11:59:58 2014 -0300
# Node ID 998d4712e9a97148325b1a26d94e268d964c78c2
# Parent  849a3afc1f352ae79adf58fdfdf31b9edf2c92bc
[Core/Players] Solve issues with charge, fear, confuse or disorient spells and
               falling under the map when teleporting to Arenas or other locations.

diff -r 849a3afc1f35 -r 998d4712e9a9 src/game/Player.cpp
--- a/src/game/Player.cpp   Mon Jun 30 01:08:42 2014 +0200
+++ b/src/game/Player.cpp   Thu Jul 10 11:59:58 2014 -0300
@@ -1593,6 +1593,11 @@

     // reset movement flags at teleport, because player will continue move with these flags after teleport
     SetUnitMovementFlags(0);
+    if (hasUnitState(UNIT_STAT_LOST_CONTROL)) 
+    {
+        StopMoving(); 
+        GetMotionMaster()->Clear();
+    }

     if (m_transport)
     {

```

Try this out and get back to me.

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** LordUsagi
**Original Date:** 12.07.2014 14:55:25 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/605/#comment-11205350

<hr>

Hello?

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** smoldar
**Original Date:** 12.07.2014 15:34:06 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/605/#comment-11205662

<hr>

Sorry i was out of pc for some days. Now i testing it. Now i reproduced this bug ...
I go to compile and try it again. Oh 4 wow process at same time :D and joining is hard.

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** LordUsagi
**Original Date:** 12.07.2014 15:37:57 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/605/#comment-11205714

<hr>

:P

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** smoldar
**Original Date:** 12.07.2014 15:38:25 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/605/#comment-11205721

<hr>

This commit not work. :( Still is going in arena. Fear moved me out ... over gate now.

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** smoldar
**Original Date:** 12.07.2014 15:39:28 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/605/#comment-11205730

<hr>

I used Howl of Terror from warlock for bug it. After of cast clicked on enter to arena (howl of terror still active in arena).

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** LordUsagi
**Original Date:** 13.07.2014 14:49:07 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/605/#comment-11213196

<hr>

Howl of Terror is the only one that this commit doesn't fix. It seems to be due to pathing and calculating final destination.

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** smoldar
**Original Date:** 13.07.2014 15:16:09 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/605/#comment-11213430

<hr>

Yes, i think about remove this fear when change map.

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** Zaffy
**Original Date:** 22.07.2014 18:57:34 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/605/#comment-11377520

<hr>

Remove debuffs on arena join, also remove fear and charm effects on teleporting to another map - closes #605

→ <<cset c468c7938ba1>>
